src/network/webtask.cpp

```diff
@@ -128,27 +128,19 @@ void WebTask::onNetworkError(
 
     DEBUG_ASSERT(errorCode != QNetworkReply::NoError);
     switch (errorCode) {
-    case QNetworkReply::OperationCanceledError:
-        // Client-side abort or timeout
-        m_state = State::Aborted;
-        break;
-    case QNetworkReply::TimeoutError:
-        // Network or server-side timeout
+    case QNetworkReply::OperationCanceledError: // Client-side timeout
+    case QNetworkReply::TimeoutError:           // Network or server-side timeout
         m_state = State::TimedOut;
         break;
     default:
         m_state = State::Failed;
     }
     DEBUG_ASSERT(hasTerminated());
 
-    if (m_state == State::Aborted) {
-        emitAborted(responseWithContent.requestUrl());
```

Comment:
This condition can no longer be true because we have removed `m_state = State::Aborted;` above. 

The original issue was that emitAborted() is designed to be the responds to manual user Abort(). There is no change in the GUI. Now we emitNetworkError() in the client timeout case as well.

Reply:
So does the Abort state have any purpose then? Can we remove it?

Reply:
We need it to change the state from any other state when calling slotAbort()

However we have some other places where State::Abort is used in responds of an error. I will do some checks if this even works as desired.

Reply:
Done.